changelog/james-prysm_fix-optimistic-check.md

```diff
@@ -0,0 +1,3 @@
+### Fixed
+
+- fixes e2e introduced by PR#14997 by changing the optimistic check in getblockroot for rest APIs.
```

Comment:
I would change this as this actually fixes two existing bugs. The failing test must be wrong or have a bad assumption on head. The Sync committee production is a bad bug as well, needs to be commented here with the fix.